### ISSUE TITLE
Add ability to access Tabulator scope from Ajax ajaxURLGenerator callback

### ISSUE THREAD

**Issue by nl095059** [olifolkerd/tabulator#3017]
### Is your feature request related to a problem? Please describe.*
When using the ajaxURLGenerator in the ajax module to create our own URLs for the data fetch, it would be nice to be able to access the Tabulator scope in a similar way to the other AJAX functions.

The issue we currently have is as follows. We initiate a query external to tabulator and receive an query token. We would then like to set the query token on the Tabulator object and then use it within ajaxURLGenerator. 

### Describe the solution you'd like
Looking at the code, it appears the only change needed would be as follows:

```
-  url = self.urlGenerator(url, config, params);
+ url = self.urlGenerator.call(self.table, url, config, params);
```

and would fall in line with how the ajaxReponse and ajaxError are working. Happy to raise a PR if you feel this is something that you would want and wouldn't have a detrimental effect to existing users.

**Comment by olifolkerd** [comment]
Great idea, please submit a PR, i would be happy to approve it.

Cheers

Oli :)

**Comment by harbulot** [comment]
I'm not using it myself, but it might be better to keep the first 3 parameters as they are and add the new one as a 4th. Otherwise, this change will break the code of users who've relied on `url, config, params` being the first 3.

**Comment by olifolkerd** [comment]
@harbulot you have misunderstood the proposed update there.

when using the `call` function the first argument is the scope of the function, which is then followed by the actual aguments of the function.

This change will not alter the order of the arguments of the called function in any way, it will simply change the scop of the this variable inside the function

Cheers

Oli :)

**Comment by harbulot** [comment]
Sorry, my mistake :-/

**Comment by olifolkerd** [comment]
This has now been merged into the 4.9 release

Cheers

Oli :)
